### DISCUSSION TITLE
how do I fix this?

### DISCUSSION THREAD

**Discussion by MrMidnightOfficial** [vlang/v#14792] [General]
how do I fix this? I got a error message and the output was:

./test.v:153:36: error: to propagate the optional call, `gen_exitsing` must return an optional
  151 | fn gen_exitsing() {
  152 | a := ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
  153 | mut ghtrhtrh := a[ rand.intn(a.len)? ]
      |                                    ^
  154 | println(ghtrhtrh)
  155 | }

and here is the fn:
`fn gen_exitsing() {`
`a := ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']`
`mut ghtrhtrh := a[ rand.intn(a.len)? ]`
`println(ghtrhtrh)`
`}`

**Comment by larpon** [comment]
Your `gen_exitsing` doesn't return an optional.
In V you need to handle errors on the spot or *propagate* them to the caller.
Either handle the optional return that `rand.intn` can return (with an `or {}` block) or make `gen_exitsing` return an optional.
https://github.com/vlang/v/blob/master/doc/docs.md#optionresult-types-and-error-handling

**Reply by MrMidnightOfficial** [reply]
what is an optional?

**Reply by JalonSolov** [reply]
Read the link he gave you.

**Comment by changrui** [comment]
`fn gen_exitsing() ? {`
